In [1]:
from __future__  import print_function, division
import pylab as pl
import pandas as pd
import numpy as np
import os

%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [4]:
df = pd.read_csv(os.getenv("PUIDATA") + "/" + 'times.txt', delimiter="\t", names = ['time'])

In [5]:
df.head()

time
0  31.622239
1  32.821376
2  30.229101
3  31.413766
4  39.010550

## _$H_0$_: $T_{new}$ >= $T_{old}$

## _$H_1$_: $T_{new}$ < $T_{old}$

In [14]:
u = df.time.mean();u

34.466161688299998

In [16]:
N = df.time.count(); N

100

In [27]:
Z = (36 - u) / (6 / np.sqrt(N)); Z

2.5563971861666701

A z-score of 2.6 is larger than 2, which mean the new mean is 2 sigmas away from the old mean. 2 sigma is a 5% threshold. 2.6 z-score passes the 5% threshold and, therefore, rejects the null hypothesis.